src/app/shared/components/button/button.component.ts

```diff
@@ -1,29 +1,50 @@
-import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
-import { IconProp, SizeProp } from '@fortawesome/fontawesome-svg-core'
-import { IconButton, IconButtonSize } from '../../enums/icon-button'
-import { IconService } from 'src/app/core/abstractions/icon.service'
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core'
 
 @Component({
   selector: 'sofie-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ButtonComponent implements OnInit {
-  @Input() public iconButtonSize: IconButtonSize
-  @Input() public iconButton: IconButton
-  @Input() public label: string = ''
-  @Input() public selected: string = ''
-  @Output() public readonly onClick: EventEmitter<void> = new EventEmitter<void>()
-  constructor(private readonly iconService: IconService) {}
+export class ButtonComponent {
+  @Input() public disabled: boolean = false
+  @Input() public type: 'STANDARD' | 'PRIMARY' | 'WARNING' | 'DANGER' = 'STANDARD'
 
-  public iconButtonSizeProp: SizeProp
-  public iconButtonProp: IconProp
+  constructor() {}
 
-  public ngOnInit(): void {
-    if (this.iconButton && this.iconButtonSize) {
-      this.iconButtonProp = this.iconService.getIconProperty(this.iconButton)
-      this.iconButtonSizeProp = this.iconButtonSize ? this.iconButtonSize : IconButtonSize.M
+  get getBackgroundColor(): string {
+    switch (this.type) {
+      case 'WARNING': {
+        return 'var(--yellow-color)'
+      }
+      case 'DANGER': {
+        return 'var(--danger-color)'
+      }
+      case 'PRIMARY': {
+        return 'var(--blue-color)'
+      }
+      case 'STANDARD':
+      default: {
+        return 'var(--white-color)'
+      }
+    }
+  }
+
+  get getTextColor(): string {
+    switch (this.type) {
+      case 'WARNING': {
+        return 'var(--white-color)'
+      }
+      case 'DANGER': {
+        return 'var(--white-color)'
+      }
+      case 'PRIMARY': {
+        return 'var(--white-color)'
+      }
+      case 'STANDARD':
+      default: {
+        return 'var(--black-color)'
+      }
     }
```

Comment:
If the `type` property gets the host binding decorator `@HostBinding('class')`, then all of the styling can be moved to the SCSS file.

Reply:
I would like to see an example of this :)

Reply:
I have created an example without using host binding, since we want to target the button element inside the host element. I have created an example where the button files looks like:
```typescript
// button.component.ts
import { ChangeDetectionStrategy, Component, Input } from '@angular/core'

@Component({
  selector: 'sofie-button',
  templateUrl: './button.component.html',
  styleUrls: ['./button.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ButtonComponent {
  @Input() public isDisabled: boolean = false
  @Input() public type: 'STANDARD' | 'PRIMARY' | 'WARNING' | 'DANGER' = 'STANDARD'
}
```
```html
// button.component.html
<button
  class="button p-2 pe-3 ps-3"
  [disabled]="isDisabled"
  [ngClass]="type.toLowerCase()"
  >
  <ng-content></ng-content>
</button>
```
```scss
// button.component.scss
.button {
  cursor: pointer;

  font-size: 1rem;

  border-radius: 4px;
  border: 1px solid var(--gray-color);

  color: var(--black-color);
  background-color: var(--white-color);

  &:hover {
    filter: brightness(95%);
  }

  &:disabled,
  &[disabled] {
    cursor: auto;
    filter: brightness(70%);
  }

  &.danger {
    background-color: var(--danger-color);
    color: var(--white-color);
  }

  &.warning {
    background-color: var(--attention-color);
    color: var(--white-color);
  }

  &.primary {
    background-color: var(--blue-color);
    color: var(--white-color);
  }
}
```

The result is:
<img width="335" alt="Skærmbillede 2024-03-20 kl  21 21 53" src="https://github.com/tv2/ng-sofie/assets/11258272/722687a1-31bd-4096-87ea-48ef09bbb09b">